### ISSUE TITLE
Add support for BlobAccess to accept environment

### ISSUE THREAD

**Issue by mkalish** [CDCgov/prime-reportstream#10484]
### User Story
As a report stream developer, I would like to choose the environment I am accessing azure blob storage for

### Description/Use Case
To facilitate future work we need to add the ability for BlobAccess to access environments other than the current RS instance. Add a means to allow BlobAccess to target a configurable environment. 

### Dev Notes
The tech debt portion of this ticket was spun off to #10774 

This work could take several forms, such as:
BlobAccess should take `Environment` in its constructor, then be able to access blobs for that specific environment.
or
BlobAccess should take `Environment` as an optional parameter in function calls, then be able to access blobs for that specific environment.

### Acceptance Criteria
- [ ] BlobAccess is able to access blobs in other environments

**Comment by bishoyayoub** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Platform-6182b02547c1130010f459db/issues/CDCgov/prime-reportstream/10484?planning-poker) with Zenhub @arnejduranovic @jimmyfagan @JFU-NAVA-PBC @mkalish @luis-pabon-tf @jack-h-wang @JessicaWNava

**Comment by jack-h-wang** [comment]
For discussion: BlobAccess is largely a collection of static methods, with `uploadReport` being the only exception. Does it make sense to instantiate an instance of BlobAccess per environment, or should environment be a parameter for each function, keeping BlobAccess itself agnostic?

**Comment by arnejduranovic** [comment]
@jeremy-page is this blocked by CDC Security? @jack-h-wang
